src/containers/pages/InterventionPlan/IRS/plan/index.tsx

```diff
@@ -539,6 +591,57 @@ class IrsPlan extends React.Component<
     );
   }
 
+  /** getAllAssignments - funciton to get all assignments for the plan from OpenSRP
+   * @param {string} planId the id of the plan for which to return assignments
+   * @param {Organization[]} organizations an array of Organizations by which to query assignments
+   * @param {(assignments:Assignment[])=>any} callback a function handling the resulting Assignments
+   * @returns {(assignments:Assignment[])=>any} the callback passing the Assignments as the first parameter
+   */
+  public getAllAssignments(
+    planId: string,
+    organizations: Organization[],
+    callback: (assignments: Assignment[]) => any
+  ) {
+    // define a reference of all organization ids
+    const organizationIds: string[] = organizations.map((o: Organization) => o.identifier);
+
+    // generate api calls for each organization id
+    const assignmentPromises = organizationIds.map(
+      (o: string) =>
+        new Promise((resolve, reject) => {
+          OpenSrpAssignedService.read(o, { plan: planId })
+            .then(j => resolve(j))
+            .catch(error => reject(error));
+        })
+    );
+
+    // make all the calls to get assignments per org for the plan
+    return Promise.all(assignmentPromises).then((results: any[]) => {
+      const nextAssignments: Assignment[] = [];
+      // step through all api results per organization
+      for (let o = 0; o < results.length; o += 1) {
+        const result = results[o];
+        if (Array.isArray(result)) {
```

Comment:
you could also pre-filter and remove the `if` from the loop body,

```javascript
results.filter(result => Array.isArray(result)).forEach(...
```

but that'd be slower since it'd required an additional list iteration, so actually don't do that